component/src/main/java/io/siddhi/extension/map/protobuf/sinkmapper/ProtobufSinkMapper.java

```diff
@@ -277,7 +278,16 @@ public void mapAndSend(Event event, OptionHolder optionHolder, Map<String, Templ
             Method buildMethod = messageBuilderObject.getClass().getDeclaredMethod(GrpcConstants.BUILD_METHOD);
             Object messageObject = buildMethod.invoke(messageBuilderObject); //get the message object by invoking
             // build() method
-            sinkListener.publish(messageObject);
+            if (sinkType.startsWith(GrpcConstants.GRPC_PROTOCOL_NAME)) {
```

Comment:
Do a uppercase/lowercase and then check startsWith. Else, it will be case sensitive